odoo_test_xmlrunner/odoo_tests/loader.py

```diff
@@ -0,0 +1,33 @@
+import os
+import threading
+
+import xmlrunner
+
+from odoo.tests import loader as odoo_loader
+from odoo.tests.result import OdooTestResult
+from odoo.tools import config
+
+
+def new_run_suite(suite, module_name=None):
+    # Override : Get and create a config dir
+    test_result_directory = config.get("test_result_directory", "test_results")
+    # create test result directory if not exists
+    if not os.path.exists(test_result_directory):
+        os.makedirs(test_result_directory)
+
+    # avoid dependency hell
+    from odoo.modules import module
+
+    module.current_test = module_name
+    threading.current_thread().testing = True
+    results = OdooTestResult()
```

Comment:
This returns an empty `results` which leads to displaying no test results and fails to exit the process with the correct code. Please have a look to my backport : https://github.com/OCA/server-tools/pull/3127